lib/selenium-cucumber/methods/click_elements_methods.rb

```diff
@@ -1,10 +1,23 @@
 require_relative 'required_files'
 
-
 def click(access_type, access_name)
   $driver.find_element(:"#{access_type}" => "#{access_name}").click
 end
 
+def click_by_text(access_type, access_name, text)
```

Comment:
I got the idea and its good thought,  I was trying to improve logic take following e.g
You have 1000 links in a page with same class and unique text for each.

scenario 1 : you passed text which is not present then you end-up with iterating 1000 links, so better to check does the text present in page source and then only enter in loop or find elements.

scenario 1 : you passed text which is of last link you will again run loop for 1000 times. but if you pass with xpath like
//a[@class='abc' and text()='xyz'] it will target only that particular element.